src/action/create_collection.rs

```diff
@@ -12,10 +12,10 @@ impl Database {
     ///
     /// `await` will return d[`Result<()>`].
     #[deeplink]
-    pub fn create_collection(&self, name: impl AsRef<str>) -> CreateCollection {
+    pub fn create_collection(&self, name: impl Into<String>) -> CreateCollection {
```

Comment:
This seems more idiomatic when we're converting into an owned `String` -- should we do this across the board? I think there are a few more inputs we should change outside the changes in this PR (e.g. the `WriteConcern` construction methods), but I can just file a separate ticket to do a full audit

Reply:
Yeah, I think a ticket for a general pass would make sense.  I did this here because I realized `name.as_ref().to_owned()` was silly.